Examples/RegistrationITKv4/MultiResImageRegistration1.cxx

```diff
@@ -666,3 +660,25 @@ main(int argc, const char * argv[])
 
   return EXIT_SUCCESS;
 }
+} // namespace
+int
+main(int argc, const char * argv[])
+{
+  try
+  {
+    return ExampleMain(argc, argv);
+  }
+  catch (const itk::ExceptionObject & exceptionObject)
+  {
+    std::cerr << "ITK exception caught:\n" << exceptionObject << '\n';
+  }
+  catch (const std::exception & stdException)
+  {
+    std::cerr << "std exception caught:\n" << stdException.what() << '\n';
+  }
+  catch (...)
+  {
+    std::cerr << "Unhandled exception!\n";
+  }
```

Comment:
I hesitate because this approach still may not keep [clang-tidy bugprone-exception-escape](https://clang.llvm.org/extra/clang-tidy/checks/bugprone/exception-escape.html) silent. At least _in theory_, `std::cerr << "..."` might still throw an exception.

This could be solved by using a C function like `fputs("...", stderr)`, instead of `std::cerr`. Or by adding an extra `try` `catch(...)`.

On the other hand, we could also decide to just ignore such clang-tidy checks.

Reply:
Are there any known situations in which `std::cerr << "...";` throws an exception?

Reply:
To use `fputs` we would need to `#include <stdio.h>`. Inconvenient. I am not sure we want to teach this practice in examples.

Reply:
I am not regularly running clang-tidy, so it is easy for me to ignore its objections 😄 I vote for this option.

Reply:
Thanks @dzenanz I'm neutral, if that's also an option  😺